### PR TITLE
[FlagGems Operator Development Competition] Add max_pool3d operator

### PR DESCRIPTION
**Pull request by Potato-wll**
### PR Category
Operator

### Type of Change
New Feature

### Description

Add `max_pool3d` operator implementation using Triton kernels, including forward pass, backward pass, and index tracking.

**Implemented operators:**
- `max_pool3d`: High-level API matching `torch.nn.functional.max_pool3d`, supporting `return_indices` parameter.
- `max_pool3d_with_indices`: Forward pass that returns both output and max indices (IR-level op).
- `max_pool3d_backward`: Backward pass using saved indices for gradient computation (IR-level op).

**Key features:**
- Triton kernels with `@triton.autotune` for both forward (`max_pool3d_forward_kernel`) and backward (`max_pool3d_backward_kernel`) passes, supporting multiple block configurations for different input shapes.
- Full parameter support: `kernel_size`, `stride`, `padding`, `dilation`, `ceil_mode`, `return_indices`, each accepting both `int` and `tuple[int, int, int]` forms.
- Compatible with both 4D `(C, D, H, W)` and 5D `(N, C, D, H, W)` input tensors.
- Input validation with clear error messages for invalid parameters.

**Changes:**
- `src/flag_gems/ops/max_pool3d_with_indices.py` — New operator implementation (576 lines).
- `src/flag_gems/ops/__init__.py` — Register new operators in module exports.
- `src/flag_gems/__init__.py` — Register operators in global config.
- `conf/operators.yaml` — Add operator metadata entries.
- `tests/test_reduction_ops.py` — Add 28 parameterized test configurations covering forward, return_indices, and backward accuracy.
- `benchmark/test_reduction_perf.py` — Add forward and backward performance benchmarks.

### Issue

- Associated with [FlagGems Operator Development Competition - max_pool3d (中级)](https://flagos.io/RaceDetail?id=296flq8k&lang=cn)

### Progress

- [x] Change is responded to an issue.
- [x] Change is fully covered by a UT.
- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).

**Accuracy Test Results:**
```python
pytest -m max_pool3d -v
```
image
image
image
image

### Performance

Benchmark tests are included in `benchmark/test_reduction_perf.py` via `test_perf_max_pool3d` and `test_perf_max_pool3d_backward`, using the FlagGems `GenericBenchmark` framework with 5D input shapes of various scales:
- `(2, 3, 16, 32, 32)`, `(4, 16, 8, 28, 28)`, `(8, 32, 8, 16, 16)`, `(2, 64, 4, 14, 14)`, `(1, 128, 4, 8, 8)`

Comprehensive benchmark level additionally tests non-cubic kernel/stride/padding, dilation, and ceil_mode variants.

```python
pytest benchmark/test_reduction_perf.py -m max_pool3d -s
```
image
image
image
image
image

### Test Coverage

28 parameterized configurations across 3 test functions (`test_accuracy_max_pool3d`, `test_accuracy_max_pool3d_return_indices`, `test_accuracy_max_pool3d_backward`) covering:

| Dimension       | Coverage                                           |
|-----------------|---------------------------------------------------|
| Input scale     | Small (2×2×2) / Regular (8×16×16) / Large (32×32×32, 16×64×64) |
| Input rank      | 4D (C,D,H,W) and 5D (N,C,D,H,W)                  |
| kernel_size     | 1, 2, 3, non-cubic tuples (e.g. (1,2,3), (2,3,3), (3,3,5)) |
| stride          | None (default=kernel_size), 1, 2, 3, non-cubic tuples |
| padding         | 0, 1, non-cubic tuples                            |
| dilation        | 1, 2, non-cubic tuples                            |
| ceil_mode       | True / False                                       |
| Combined        | Non-cubic kernel + ceil_mode + padding              |
| Data types      | float16, float32, bfloat16                         |

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
